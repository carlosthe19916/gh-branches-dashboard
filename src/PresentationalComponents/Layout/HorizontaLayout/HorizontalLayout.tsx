import * as React from "react";
import { Page, SkipToContent } from "@patternfly/react-core";
import { ButtonAboutProject } from "../ButtonAboutProject";
import { HeaderProject } from "../HeaderProject";

export interface HorizontalLayoutProps {}

interface State {}

export class HorizontalLayout extends React.Component<
  HorizontalLayoutProps,
  State
> {
  renderAboutButton = () => {
    return <ButtonAboutProject />;
  };

  render() {
    const { children } = this.props;

    const pageId = "main-content-page-layout-horizontal-nav";
    const PageSkipToContent = (
      <SkipToContent href={`#${pageId}`}>Skip to content</SkipToContent>
    );

    return (
      <React.Fragment>
        <Page
          header={<HeaderProject aboutButton={this.renderAboutButton()} />}
          skipToContent={PageSkipToContent}
          mainContainerId={pageId}
        >
          {children}
        </Page>
      </React.Fragment>
    );
  }
}
