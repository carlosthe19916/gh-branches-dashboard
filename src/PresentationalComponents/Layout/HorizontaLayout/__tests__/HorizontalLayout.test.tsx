import React from "react";
import { shallow } from "enzyme";
import { HorizontalLayout } from "../HorizontalLayout";

it("Test snapshot", () => {
  const wrapper = shallow(<HorizontalLayout />);
  expect(wrapper).toMatchSnapshot();
});
