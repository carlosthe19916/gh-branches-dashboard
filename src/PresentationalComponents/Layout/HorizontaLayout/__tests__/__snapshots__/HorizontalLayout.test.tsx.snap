// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test snapshot 1`] = `
<Fragment>
  <Page
    breadcrumb={null}
    className=""
    defaultManagedSidebarIsOpen={true}
    header={
      <HeaderProject
        aboutButton={<ButtonAboutProject />}
      />
    }
    isManagedSidebar={false}
    mainContainerId="main-content-page-layout-horizontal-nav"
    onPageResize={[Function]}
    sidebar={null}
    skipToContent={
      <SkipToContent
        className=""
        component="a"
        href="#main-content-page-layout-horizontal-nav"
        show={false}
      >
        Skip to content
      </SkipToContent>
    }
  />
</Fragment>
`;
